Vulkan-Hpp assumes that certain __cpp macros exist

When using the Vulkan C++ headers with an older compiler (in my case: clang 17), it fails to compile with the following message:

```
_deps/vulkanheaders-src/include/vulkan/vulkan_hpp_macros.hpp:82:17: error: '__cpp_lib_three_way_comparison' is not defined, evaluates to 0 [-Werror,-Wundef]
#if ( 201907 <= __cpp_lib_three_way_comparison ) && __has_include( <compare> ) && !defined( VULKAN_HPP_NO_SPACESHIP_OPERATOR )

_deps/vulkanheaders-src/include/vulkan/vulkan_to_string.hpp:25:7: error: '__cpp_lib_format' is not defined, evaluates to 0 [-Werror,-Wundef]
#  if __cpp_lib_format
```

I think that this should use `if defined(macro) && macro` to avoid these issues.

Update: It seems to be critical here that I'm using the Vulkan headers via FetchContent, which causes them to be included using `-I`. When using `-isystem` as when included via regular find_package, there are no problems. Also note that I'm using `-Werror=undef`. I guess `isystem` causes the compiler to normally hide these problems.

--- a/snippets/macros.hpp
+++ b/snippets/macros.hpp
@@ -69,11 +69,11 @@
 #  define __has_include( x ) false
 #endif
 
-#if ( 201907 <= __cpp_lib_three_way_comparison ) && __has_include( <compare> ) && !defined( VULKAN_HPP_NO_SPACESHIP_OPERATOR )
+#if defined( __cpp_lib_three_way_comparison ) && ( 201907 <= __cpp_lib_three_way_comparison ) && __has_include( <compare> ) && !defined( VULKAN_HPP_NO_SPACESHIP_OPERATOR )
 #  define VULKAN_HPP_HAS_SPACESHIP_OPERATOR
 #endif
 
-#if ( 201803 <= __cpp_lib_span )
+#if defined( __cpp_lib_span ) && ( 201803 <= __cpp_lib_span )
 #  define VULKAN_HPP_SUPPORT_SPAN
 #endif
 

--- a/vulkan/vulkan_hpp_macros.hpp
+++ b/vulkan/vulkan_hpp_macros.hpp
@@ -79,11 +79,11 @@
 #  define __has_include( x ) false
 #endif
 
-#if ( 201907 <= __cpp_lib_three_way_comparison ) && __has_include( <compare> ) && !defined( VULKAN_HPP_NO_SPACESHIP_OPERATOR )
+#if defined( __cpp_lib_three_way_comparison ) && ( 201907 <= __cpp_lib_three_way_comparison ) && __has_include( <compare> ) && !defined( VULKAN_HPP_NO_SPACESHIP_OPERATOR )
 #  define VULKAN_HPP_HAS_SPACESHIP_OPERATOR
 #endif
 
-#if ( 201803 <= __cpp_lib_span )
+#if defined( __cpp_lib_span ) && ( 201803 <= __cpp_lib_span )
 #  define VULKAN_HPP_SUPPORT_SPAN
 #endif
 

